src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -557,21 +556,11 @@ private void removeAirNoLongerInTerritory() {
         && (returnFireAgainstDefendingSubs() != ReturnFire.ALL || returnFireAgainstAttackingSubs() != ReturnFire.ALL)) {
       steps.add(REMOVE_SNEAK_ATTACK_CASUALTIES);
     }
-    // Air only Units can't attack subs without Destroyers present
-    if (isAirAttackSubRestricted()) {
-      final Collection<Unit> units = new ArrayList<>(m_attackingUnits.size() + m_attackingWaitingToDie.size());
-      units.addAll(m_attackingUnits);
-      if (Match.anyMatch(m_attackingUnits, Matches.unitIsAir()) && !canAirAttackSubs(m_defendingUnits, units)) {
-        steps.add(SUBMERGE_SUBS_VS_AIR_ONLY);
-      }
-    }
-    // Air Units can't attack subs without Destroyers present
-    if (m_battleSite.isWater() && isAirAttackSubRestricted()) {
-      final Collection<Unit> units = new ArrayList<>(m_attackingUnits.size() + m_attackingWaitingToDie.size());
-      units.addAll(m_attackingUnits);
-      if (Match.anyMatch(m_attackingUnits, Matches.unitIsAir()) && !canAirAttackSubs(m_defendingUnits, units)) {
-        steps.add(AIR_ATTACK_NON_SUBS);
-      }
+    // Air units can't attack subs without Destroyers present
+    if (isAirAttackSubRestricted() && Match.anyMatch(m_attackingUnits, Matches.unitIsAir())
+        && !canAirAttackSubs(m_defendingUnits, m_attackingUnits)) {
+      steps.add(SUBMERGE_SUBS_VS_AIR_ONLY);
+      steps.add(AIR_ATTACK_NON_SUBS);
```

Comment:
Merged these 2 methods together and simplified.

Reply:
Is `m_battleSite.isWater()` always `true`?  Or is this requirement not really necessary to add the `AIR_ATTACK_NON_SUBS` step?

Reply:
It is actually already checked in the canAirAttackSubs method.